src/backend/joanie/tests/core/test_flows_order.py

```diff
@@ -1377,3 +1413,23 @@ def test_flows_order_pending(self):
                 order = factories.OrderFactory(state=state)
                 order.flow.pending()
                 self.assertEqual(order.state, enums.ORDER_STATE_PENDING)
+
+    def test_flows_order_update(self):
+        """
+        Test that updating flow is transitioning as expected for all states.
+        """
+
+        def get_target_state(state):
+            if state == enums.ORDER_STATE_ASSIGNED:
+                return enums.ORDER_STATE_TO_SAVE_PAYMENT_METHOD
+            return state
+
+        transitions = [
+            (state, get_target_state(state)) for (state, _) in enums.ORDER_STATE_CHOICES
+        ]
+
+        for [source, target] in transitions:
+            with self.subTest(source=source, target=target):
+                order = factories.OrderGeneratorFactory(state=source)
+                order.flow.update()
+                self.assertEqual(order.state, target)
```

Comment:
```suggestion
        for (state, _) in enums.ORDER_STATE_CHOICES:
            with self.subTest(source=source, target=target):
                order = factories.OrderGeneratorFactory(state=source)
                order.flow.update()
                
                # As assinged state is transitional only, flow.update should change the state.
                # It could go to ORDER_STATE_TO_SAVE_PAYMENT_METHOD, TO_SIGN, or COMPLETED.
                # With the actual impmentation of OrderGeneratorFactory, it goes to ORDER_STATE_TO_SAVE_PAYMENT_METHOD
                if state == enums.ORDER_STATE_ASSIGNED:
                    self.assertEqual(order.state, enums.ORDER_STATE_TO_SAVE_PAYMENT_METHOD)
                else:
                    self.assertEqual(order.state, state)
```

For simplicity, and clarity.